src/Controller/UserController.php

```diff
@@ -28,21 +28,50 @@ public function register(Request $request)
         $entityManager = $this->getDoctrine()->getManager();
 
         $username=$request->query->get('username');
-        $checkUsername = $entityManager->getRepository('App:User')->findOneBy(array('username'=> $username));
-        if($checkUsername){
-            return new Response('Username already used');
+        if($username){
+            $checkUsername = $entityManager->getRepository('App:User')->findOneBy(array('username'=> $username));
+            if($checkUsername){
+                return new Response('Username already used');
+            }
+            else{
+                $password=$request->query->get('password');
+                $password_val=$request->query->get('password_validation');
+                $user=$request->query->get('username');
+                $email=$request->query->get('email');
+            }
         }
         else{
-            $password=$request->query->get('password');
-            $password_val=$request->query->get('password_validation');
-            $user=$request->query->get('username');
-            $email=$request->query->get('email');
+            return new Response('Username cant be empty');
         }
         $entityManager->flush();
-        if(strlen($user) <= 20 and strlen($password) >= 8 and filter_var($email, FILTER_VALIDATE_EMAIL) and $password==$password_val){
+        if(strlen($user) <= 20){
```

Comment:
Mabye do it as a Validation class. I will make a example in your branch, so that you can implement the rest :)

Reply:
Sure good point. Working on that.